### ISSUE TITLE
Figure out helmet config for stage/prod

### ISSUE THREAD

**Issue by pdehaan** [mozilla/idea-town#64]
Ref #60

> 9:46 AM <_6a68> I'm fine to merge like that for dev, if we file a big to figure out the stage/prod config later
> 9:47 AM <_6a68> I figure CSP will require some config, especially if we use the mozcdn

**Comment by 6a68** [comment]
thanks for opening this @pdehaan!

Yeah, CSP configuration will require getting some info from ops about where we'll be serving our assets from, and that'll vary in stage and prod. We'll want to be sure CSP works in both places.

Other default helmet behavior to worry about:
- `noCache` disables all client-side caching. I don't think we want that in production, though we might not get a lot of repeat viewers, so the primed-cache experience might not matter
- public key pinning seems maybe not worth it, because we don't normally coordinate with ops around cert updates (but maybe fxa does, and maybe we could start, just seems like work)
- `noSniff` disables mime-type sniffing. I guess we just have to make sure we're setting every mime type correctly? This might be a great thing to handle in an automated test.

We will need to get some kind of security review before this ships, so we'll need our security person to offer feedback on helmet and anything we need to do in addition. I'll open a separate bug for that and link to this one.

**Comment by pdehaan** [comment]
Here's what we used in **fxa-content-server**: https://github.com/mozilla/fxa-content-server/search?utf8=%E2%9C%93&q=helmet

**TL;DR:**

```js
var cspMiddleware = helmet.csp({
  defaultSrc: [
    SELF
  ],

  connectSrc: [
    SELF,
    getOrigin(config.get('fxaccount_url')),
    config.get('oauth_url'),
    config.get('profile_url'),
    config.get('marketing_email.api_url')
  ],

  imgSrc: [
    SELF,
    DATA,
    GRAVATAR,
    config.get('profile_images_url')
  ],

  reportUri: config.get('csp.reportUri'),
  reportOnly: config.get('csp.reportOnly')
});
```
&mdash; via [**mozilla/fxa-content-server** ./server/lib/csp.js#L27-L49](https://github.com/mozilla/fxa-content-server/blob/e4c023f754d8a1bddbbb61fea9fb9580e5e4a297/server/lib/csp.js#L27-L49)

---

```js
  app.use(helmet.xframe('deny'));
  app.use(helmet.xssFilter());
  app.use(helmet.hsts({
    maxAge: config.get('hsts_max_age'),
    includeSubdomains: true,
    force: true
  }));
  app.use(helmet.nosniff());

  if (config.get('csp.enabled')) {
    app.use(csp);
  }
```
&mdash; via [**mozilla/fxa-content-server** ./server/bin/fxa-content-server.js#L67-L78](https://github.com/mozilla/fxa-content-server/blob/660136152e3521aee937fad647f0508352bbd146/server/bin/fxa-content-server.js#L67-L78)

**Comment by pdehaan** [comment]
@edmoz to find out about getting a sec-review.
